tests/examples/test_examples.py

```diff
@@ -56,6 +56,14 @@ def replace_mlflow_with_dev_version(yml_path):
         f.write(new_src)
 
 
+@pytest.fixture(scope="function", autouse=True)
+def clean_envs_and_cache():
+    yield
+
+    if get_free_disk_space() < 7.0:  # unit: GiB
+        process.exec_cmd(["./dev/remove-conda-envs.sh"])
```

Comment:
if the free disk space is less than  7.0 GiB, run `./dev/remove-conda-envs.sh` which cleans up existing conda envs + unused package cache.